sys/include/color.h

```diff
@@ -104,6 +105,22 @@ void color_str2rgb(const char *str, color_rgb_t *color);
  */
 void color_rgb2str(const color_rgb_t *rgb, char *str);
 
+/**
+ * @brief Invert a given rgb color
+ *
+ * @param[in] rgb       Input rgb color, that should be converted.
+ * @param[out] inv_rgb  Output rgb color, result of the conversion.
+ */
+void color_rgb_invert(const color_rgb_t *rgb, color_rgb_t *inv_rgb);
+
+/**
+ * @brief Calculate the complementary color of a given rgb color
+ *
+ * @param[in] rgb       Input rgb color.
+ * @param[out] comp_rgb Output rgb color, result of the complementary color calculation.
+ */
+void color_rgb_complementary(const color_rgb_t *rgb, color_rgb_t *comp_rgb)
```

Comment:
For main colors in rgb space inversion and complementary is actually the same, with secoundary colors it is not. The code is not the HSL complementary, I basicly do what adobe does xD

https://helpx.adobe.com/de/illustrator/using/adjusting-colors.html#WS714a382cdf7d304e7e07d0100196cbc5f-6288a